cxx-squid/src/main/java/org/sonar/cxx/preprocessor/CxxPreprocessor.java

```diff
@@ -127,34 +173,56 @@ public CxxPreprocessor(SquidAstVisitorContext<Grammar> context,
 
     pplineParser = CppParser.create(conf);
 
+    final List<String> configuredDefines = conf.getDefines();
```

Comment:
@ivangalkin ctor code is difficult to read. Maybe you can split it into smaller methods.

Reply:
yes, I'll do some refactoring; I'll remove the 'try{...}finally {}' code from the ctor. It doesn make much sense here. Exceptions are not handled -> `operator new` doesn't create any object instance -> there will be no sense in house-keeping. The case in CxxPreprocessor::init() is different: there is an object, it is reused for all the processed files, the object must be kept in consistent state.